File: Data Structures & Algorithms/climbing-stairs/submission-0.cpp
class Solution {
public:
    void dp(int n , int sum , int &ans){
        if(n == sum){
            ans+=1;
            return;
        }
        else if(sum > n){
            return;
        }

        dp(n , sum + 1 , ans);
        dp(n , sum + 2 , ans);
    }
    int climbStairs(int n) {
        if(n == 0){
            return 0;
        }

        int ans = 0;
        dp(n , 0 , ans);
        return ans;
    }
};

File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-0.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int []ans = new int[arr.length];
        int max = -1;
        for(int i = arr.length - 1 ; i >= 0 ; i--){
            if(max == -1){
                ans[i] = -1;
                max = arr[i];
                continue;
            }
            ans[i] = max;
            max = Math.max(max , arr[i]);
        }

        return ans;
    }
}